src/extensions/cp/apple/finalcutpro/init.lua

```diff
@@ -1349,67 +1340,52 @@ function App:unwatch(id)
 	return false
 end
 
+-- cp.apple.finalcutpro:_initWatchers() -> none
+-- Method
+-- Initialise all the various Final Cut Pro Watchers.
+--
+-- Parameters:
+--  * None
+--
+-- Returns:
+--  * None
 function App:_initWatchers()
 
+	if not self._watchers then
+		--log.df("Setting up Final Cut Pro Watchers...")
+		self._watchers = {}
+	end
+
 	--------------------------------------------------------------------------------
-	-- Application Watcher:
+	-- Setup Application Watcher:
 	--------------------------------------------------------------------------------
-	local watcher = application.watcher
-
-	self._active = false
-	self._appWatcher = watcher.new(
-		function(appName, eventType, appObject)
-			local event = nil
-
-			if (appName == "Final Cut Pro") then
-				if self._active == false and (eventType == watcher.activated) and self:isFrontmost() then
-					self._active = true
-					event = "active"
-				elseif self._active == true and (eventType == watcher.deactivated or eventType == watcher.terminated) then
-					self._active = false
-					event = "inactive"
+	--log.df("Setting up Application Watcher...")
+	self._appWatcher = applicationwatcher.new(
+		function(appName, eventType, application)
+			if (application:bundleID() == App.BUNDLE_ID) then
+				if eventType == applicationwatcher.activated then
```

Comment:
Quick question - you removed the check for 'isFrontmost'. Was it redundant? I have vague recollections about adding the extra check so that it would only be notifying people about being 'active' when it was the top application.